### PR TITLE
return as soon as possible

### PR DESCRIPTION
**Pull request by lalnuo**

### CODE REVIEW

File: src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java [hunk 1]
```diff
@@ -267,16 +267,14 @@ public boolean hasDeadlinePassedAt(Date time) {
...
-            try {
-                DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX");
-                Date deadlineDate = format.parse(deadline);
-                return deadlineDate.getTime() < time.getTime();
-            } catch (ParseException e) {
-                log.error("Could not parse date: {}", e);
-                return false;
-            }
-        } else {
+        if (deadline == null) return false;
```
Comment (ljleppan) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java:270]: Use braces even with one-line blocks<sup>[[1]](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)</sup>.

File: src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java [hunk 2]
```diff
@@ -267,16 +267,14 @@ public boolean hasDeadlinePassedAt(Date time) {
...
+
+        try {
+            DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX");
```
Comment (ljleppan) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java:273]: Could probably hoist this to a `private static final` or some such while we're at it.

File: src/main/java/fi/helsinki/cs/tmc/core/domain/ExerciseKey.java [hunk 3]
```diff
@@ -25,13 +25,10 @@ public ExerciseKey(String courseName, String exerciseName) {
...
     @Override
     public boolean equals(Object obj) {
-        if (obj instanceof ExerciseKey) {
-            ExerciseKey that = (ExerciseKey) obj;
-            return Objects.equals(this.courseName, that.courseName)
-                    && Objects.equals(this.exerciseName, that.exerciseName);
-        } else {
-            return false;
-        }
+        if (!(obj instanceof ExerciseKey)) return false;
```
Comment (ljleppan) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/ExerciseKey.java:28]: See above about braces

File: src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java (hunk 2 above)
Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/Exercise.java:273]: Yeah, let's make it a constant; which it is.

**Pull request closed without merging**